patches/kaggle_secrets.py

```diff
@@ -135,6 +136,21 @@ def get_gcloud_credential(self) -> str:
             else:
               raise
 
+    def set_tensorflow_credential(self, credential):
+        """Sets the credential for use by Tensorflow both in the local notebook
+        and to pass to the TPU.
+        """
+        # Write to a local JSON credentials file and set
+        # GOOGLE_APPLICATION_CREDENTIALS for tensorflow running in the notebook.
+        adc_path = os.path.join(
+            os.environ.get('HOME', '/'), 'gcloud_credential.json')
```

Comment:
If the user ran any gcloud auth commands would this file already be existing?

Reply:
I don't think it's possible to run any of the `gcloud auth` commands within the notebook (I've tried) because they're interactive. This also isn't the default location for application default credentials, so it shouldn't overwrite a file if it does exist but the user would need to reset the env variable.